oscalkit-test/methods.go

```diff
@@ -0,0 +1,322 @@
+			fmt.Print(f.Name+" "+f.Type.Name(), f.Type.Kind())
+			if f.Tag != "" {
+				fmt.Println(" " + f.Tag)
+			}
+		}
+	}
+}
+
+// ProtocolsMapping Method To Parse All The Controls From Catalog.go
+func ProtocolsMapping(check []catalog.Catalog) map[string][]string {
+
+	SecurityControls := make(map[string][]string)
+
+	for CatalogCount := 0; CatalogCount < len(check); CatalogCount++ {
+		for GroupsCount := 0; GroupsCount < len(check[CatalogCount].Groups); GroupsCount++ {
+			for ControlsCount := 0; ControlsCount < len(check[CatalogCount].Groups[GroupsCount].Controls); ControlsCount++ {
+				SecurityControls[check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Id] = append(SecurityControls[check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Id], check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Class)
+				SecurityControls[check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Id] = append(SecurityControls[check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Id], string(check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Title))
+				for SubControlsCount := 0; SubControlsCount < len(check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Subcontrols); SubControlsCount++ {
+					SecurityControls[check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Subcontrols[SubControlsCount].Id] = append(SecurityControls[check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Subcontrols[SubControlsCount].Id], check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Subcontrols[SubControlsCount].Class)
+					SecurityControls[check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Subcontrols[SubControlsCount].Id] = append(SecurityControls[check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Subcontrols[SubControlsCount].Id], string(check[CatalogCount].Groups[GroupsCount].Controls[ControlsCount].Subcontrols[SubControlsCount].Title))
+				}
+			}
+		}
+	}
+	return SecurityControls
+}
+
+// GetCatalog gets a catalog
+func GetCatalog(r io.Reader) (*catalog.Catalog, error) {
+	o, err := oscal.New(r)
+	if err != nil {
+		return nil, err
+	}
+	if o.Catalog == nil {
+		return nil, fmt.Errorf("cannot map profile")
+	}
+	return o.Catalog, nil
+}
+
+// GetProfile gets a profile
+func GetProfile(r io.Reader) (*profile.Profile, error) {
+	o, err := oscal.New(r)
+	if err != nil {
+		return nil, err
+	}
+	if o.Profile == nil {
+		return nil, fmt.Errorf("cannot map profile")
+	}
+	return o.Profile, nil
+}
+
+// controlInProfile checks if the control provided exists in the provided profile or not
+func controlInProfile(controlID string, profile []string) bool {
+	for _, value := range profile {
+		if value == controlID {
+			return true
+		}
+	}
+	return false
+}
+
+// DownloadCatalog writes the JSON of the provided URL into a catalog.json file
+func DownloadCatalog(url string) error {
+	save := strings.Split(url, "/")
+	err := os.Mkdir("./tmp", 0777)
```

Comment:
Use the built-in function for creating a temporary directory. `ioutil.TempDir()`. See https://golang.org/pkg/io/ioutil/#TempDir